src/Cake/Commands/DebugCommand.cs

```diff
@@ -0,0 +1,34 @@
+﻿using System;
+using Cake.Core.Scripting;
+using Cake.Scripting;
+
+namespace Cake.Commands
+{
+    /// <summary>
+    /// A command that builds and debugs a build script.
+    /// </summary>
+    internal sealed class DebugCommand : ICommand
+    {
+        private readonly IScriptRunner _scriptRunner;
+        private readonly DebugScriptHost _host;
+
+        // Delegate factory used by Autofac.
+        public delegate DebugCommand Factory();
+
+        public DebugCommand(IScriptRunner scriptRunner, DebugScriptHost host)
+        {
+            _scriptRunner = scriptRunner;
+            _host = host;
+        }
+
+        public bool Execute(CakeOptions options)
+        {
+            if (options == null)
+            {
+                throw new ArgumentNullException("options");
+            }
+            _scriptRunner.Run(_host, options.Script, options.Arguments);
```

Comment:
Currently it will crash if break point isn't an task as the debugger isn't attached until tasks are executed.
![image](https://cloud.githubusercontent.com/assets/1647294/15210208/a5c56c16-1835-11e6-8790-e3dc0b15a550.png)

I.e. if I start the script with a `System.Diagnostics.Debugger.Break();` at root level.

Did a quick test by adding this before the `_scriptRunner.Run(_host, options.Script, options.Arguments);`
```csharp
            if (!System.Diagnostics.Debugger.IsAttached)
            {
                System.Diagnostics.Debugger.Launch();
            }
```

This will launch the visual studio debug selector:
![image](https://cloud.githubusercontent.com/assets/1647294/15210169/64148a18-1835-11e6-8e70-be784ee44fc6.png)

Then break at the launch
![image](https://cloud.githubusercontent.com/assets/1647294/15210287/07bfc196-1836-11e6-8191-1396713c734a.png)

F5 will take you to the "root level" Break and all works from there
![image](https://cloud.githubusercontent.com/assets/1647294/15210322/2a8ee080-1836-11e6-9941-f12742c69a1b.png)

@cake-build/team & @mholo65 what do you think perhaps makes sense to move DebugScriptHost::RunTarget to DebugCommand::Execute?